streams/signature_file.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "hash_object.proto";
+
+/**
+ * The record signature file which is created for each record stream file
+ * that signs the hash of the entire corresponding stream file.
+ */
+message SignatureFile {
+
+  /**
+   * Signature for the file
+   */
+  SignatureObject file_signature = 1;
+
+  /**
+   * Metadata signature
+   */
+  SignatureObject metadata_signature = 2;
+}
+
+/**
+ * A Signature defined by its type, length, checksum and signature bytes and the hash that is signed
+ */
+message SignatureObject {
+  /**
+   * The signature type
+   */
+  SignatureType type = 1;
+
+  /**
+   * Signature length
+   */
+  int32 length = 2;
+
+  /**
+   * Signature checksum
+   */
+  int32 checksum = 3;
+
+  /**
+   * Signature bytes
+   */
+  bytes signature = 4;
+
+  /**
+   * The hash that is signed by this signature
+   */
+  HashObject hash_object = 5;
+}
+
+/**
+ * The signature type
+ */
+enum SignatureType {
+  SIGNATURE_TYPE_UNKNOWN = 0;
```

Comment:
nit:

```suggestion
  UNKNOWN = 0;
```

Reply:
I tried making each first value to be "UNKNOWN", but the IDE is giving error "'proto.UNKNOWN' is already defined in file 'hash_object.proto'". That is the reason, why the values are different.

Reply:
I consider this as a proper way of dealing with the problem:

https://github.com/protobuf-net/protobuf-net/issues/60#issuecomment-461509160

We will preserve the name, but the generated class will point to explicit value.

I will commit the change once I receive your feedback.